control/controlclient/map.go

```diff
@@ -239,6 +241,12 @@ func (ms *mapSession) updateStateFromResponse(resp *tailcfg.MapResponse) {
 
 	if resp.Node != nil {
 		ms.lastNode = resp.Node.View()
+
+		capSet := set.Set[tailcfg.NodeCapability]{}
+		for _, v := range resp.Node.Capabilities {
```

Comment:
i think you also need to go through `resp.Node.CapMap`

Reply:
existing code doesn't

Reply:
yes it does, you just deleted the check. see `hasCapability` => `n.HasCap`

Reply:
or it does